### PR TITLE
Table Molecule: option to hide header

### PR DESCRIPTION
**Pull request by ajwl**
In some designs, we hide the header of a table. This adds an option so the user can choose not to show the header

But we don't want to remove the header entirely - ie: by using `display: none;` - because column widths are calculated by the table header cells. 

Instead we apply a class that uses `visibility: hidden;` and various other CSS properties to keep the effects of the table header but to hide it from view. 

QUESTION: should i add an option to make the first row bold? or should this be achievable thru the style prop? 

WITH HEADER 
Screenshot 2024-04-03 at 12 07 54

WITHOUT HEADER 
Screenshot 2024-04-03 at 12 07 49

- works less well for sortable tables - but by definition these should have a header - otherwise they are not sortable

### CODE REVIEW

File: src/lib/components/molecules/table/index.jsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import { useState } from 'preact/hooks'
 import { useTable } from './useTable'
 import { Chevron } from '$particles/chevron'
-import defaultStyles from './style.module.css'
+import defaultStyles from './style.module.scss'
 import { mergeStyles } from '$styles/helpers/mergeStyles'
 
-export function Table({ columns, data, styles }) {
+export function Table({ columns, data, hideHeader, styles }) {
```
Comment (njdehoog) [inline src/lib/components/molecules/table/index.jsx:7]: Let's give this a default value of `false` to simplify our lives, i.e. `hideHeader = false`.

**Review by njdehoog — CHANGES_REQUESTED** [review]
Looks good. One comment about default prop value.

File: src/lib/components/molecules/table/index.jsx (hunk 1 above)
Comment (ajwl) [reply src/lib/components/molecules/table/index.jsx:7]: yes sure

**Comment by ajwl** [comment]
@njdehoog - fixed - see above!

File: src/lib/components/molecules/table/table.stories.jsx [hunk 2]
```diff
@@ -36,6 +36,7 @@ export const Default = {
         lastName: 'Smith',
       },
     ],
+    hideHeader: false,
```
Comment (njdehoog) [inline src/lib/components/molecules/table/table.stories.jsx:39]: Sorry to be pedantic, but we can now remove this prop from the stories

Comment (ajwl) [reply src/lib/components/molecules/table/table.stories.jsx:39]: oh okay - sure - well we should keep it in one at least to document the option. I'll keep it in the party example

**Pull request merged**